### PR TITLE
Update error messages in survey form

### PR DESCRIPTION
**Pull request by kr8n3r**
## What 
Updates error messages in survey form so they are more helpful.

Error message hint taken from https://design-system.service.gov.uk/patterns/email-addresses/#error-messages

Error message aligns with the feedback component's generic error message: https://github.com/alphagov/govuk_publishing_components/blob/master/app/assets/javascripts/govuk_publishing_components/components/feedback.js#L126

## Why
The feedback component in govuk_publishing components was [updated to take into account the 422 HTTP status response](https://github.com/alphagov/govuk_publishing_components/pull/918)

While the error message provided by this app for the "Is there anything wrong with this page?" form was useful, the email survey messages aren't that useful (see below)

## Before
Screen Shot 2019-06-18 at 13 10 42

## After
Invalid Email Address:

Screen Shot 2019-06-18 at 13 05 15

Service Unavailable:

Screen Shot 2019-06-18 at 13 05 04

### CODE REVIEW

**Comment by elliotcm** [comment]
The technical change looks OK (though it could be tweaked a bit).  Has this gone through content team signoff?

**Comment by kr8n3r** [comment]
is there a content team for this app?

File: spec/requests/email_survey_signup_request_spec.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@
       expect(response.content_type).to eq("application/json")
       json_response = JSON.parse(response.body)
       expect(json_response).to have_key "message"
-      expect(json_response["message"]).to eq "email survey sign up failure"
+      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to send your message as you haven’t given us a valid email address.</h2> <p>Enter an email address in the correct format, like name@example.com</p>"
```
Comment (pollygreen) [inline spec/requests/email_survey_signup_request_spec.rb]:
```suggestion
      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to send your message as you haven’t given us a valid email address.</h2> <p>Enter an email address in the correct format, like name@example.com</p>"
```

File: spec/requests/email_survey_signup_request_spec.rb [hunk 2]
```diff
@@ -70,7 +70,7 @@
       expect(response.content_type).to eq("application/json")
       json_response = JSON.parse(response.body)
       expect(json_response).to have_key "message"
-      expect(json_response["message"]).to eq "email survey sign up failure"
+      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to receive your message right now.<h2> <p>If the problem persists, we have other ways for you to provide feedback on the contact page.</p>"
```
Comment (pollygreen) [inline spec/requests/email_survey_signup_request_spec.rb]:
```suggestion
      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to receive your message right now<h2> <p>If the problem persists, you could give us feedback on the <a href="https://www.gov.uk/contact/govuk">GOV.UK contact page</a>.</p>"
```

File: spec/requests/email_survey_signup_request_spec.rb [hunk 3]
```diff
@@ -91,7 +91,7 @@
       expect(response.content_type).to eq("application/json")
       json_response = JSON.parse(response.body)
       expect(json_response).to have_key "message"
-      expect(json_response["message"]).to eq "email survey sign up failure"
+      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to send your message as you haven’t given us a valid email address.</h2> <p>Enter an email address in the correct format, like name@example.com</p>"
```
Comment (pollygreen) [inline spec/requests/email_survey_signup_request_spec.rb]:
```suggestion
      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to send your message</h2> <p>It looks like you haven't given a valid email address. Enter an email address in the correct format, like name@example.com.</p>"
```

File: spec/requests/email_survey_signup_request_spec.rb [hunk 4]
```diff
@@ -105,7 +105,7 @@
       expect(response.content_type).to eq("application/json")
       json_response = JSON.parse(response.body)
       expect(json_response).to have_key "message"
-      expect(json_response["message"]).to eq "email survey sign up failure"
+      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to receive your message right now.<h2> <p>If the problem persists, we have other ways for you to provide feedback on the contact page.</p>"
```
Comment (pollygreen) [inline spec/requests/email_survey_signup_request_spec.rb]:
```suggestion
      expect(json_response["message"]).to eq "<h2>Sorry, we’re unable to receive your message right now<h2> <p>If the problem persists, you could give us feedback on the <a href="https://www.gov.uk/contact/govuk">GOV.UK contact page</a>.</p>"
```

**Comment by chao-xian** [comment]
I'm going to refactor this to use the languages api as we have for [some bits](https://github.com/alphagov/feedback/blob/master/config/locales/en.yml).

**Comment by injms** [comment]
From the screenshot above, we need to also include an [error message](https://design-system.service.gov.uk/components/error-message/) next to the form field if the email is not valid.

I can't get the email address validation message, as the form input is `type="email"`, so can be validated by the browser (Firefox, Chrome, Safari, Edge - not IE). This is fine, as that's what [recommended for email inputs by the Design System](https://design-system.service.gov.uk/patterns/email-addresses/), just wanted to note it.

**Comment by kr8n3r** [comment]
@injms error message and error summary will need to be added in the publishing components gem i _think_

**Comment by injms** [comment]
@kr8n3r - looks like [error message](https://govuk-publishing-components.herokuapp.com/component-guide/error_message) has been added 👍

**Comment by kr8n3r** [comment]
cool, just needs to be added to the feedback component then :) I say just...

**Comment by chao-xian** [comment]
https://trello.com/c/1pgGleVG/7-2-feedback-error-messages-are-not-very-useful

**Pull request merged**